README.md

```diff
@@ -363,6 +366,213 @@ async function main() {
+
+After configuring the Azure provider to impersonate a service account, a credential configuration file needs to be generated.
+Unlike service account credential files, the generated credential configuration file will only contain non-sensitive metadata to instruct the library on how to retrieve external subject tokens and exchange them for service account access tokens.
+The configuration file can be generated by using the [gcloud CLI](https://cloud.google.com/sdk/).
+
+To generate the Azure workload identity configuration, run the following command:
+
+```bash
+# Generate an Azure configuration file.
+gcloud iam workload-identity-pools create-cred-config \
+projects/$PROJECT_NUMBER/locations/global/workloadIdentityPools/$POOL_ID/providers/$AZURE_PROVIDER_ID \
+--service-account $SERVICE_ACCOUNT_EMAIL \
+--azure \
+--output-file /path/to/generated/config.json
+```
+
+Where the following variables need to be substituted:
+- `$PROJECT_NUMBER`: The Google Cloud project number.
+- `$POOL_ID`: The workload identity pool ID.
+- `$AZURE_PROVIDER_ID`: The Azure provider ID.
+- `$SERVICE_ACCOUNT_EMAIL`: The email of the service account to impersonate.
+
+This will generate the configuration file in the specified output file.
+
+You can now [start using the Auth library](#using-external-identities) to call Google Cloud resources from Azure.
+
+### Accessing resources from an OIDC identity provider
+
+In order to access Google Cloud resources from an identity provider that supports [OpenID Connect (OIDC)](https://openid.net/connect/), the following requirements are needed:
+- A workload identity pool needs to be created.
+- An OIDC identity provider needs to be added in the workload identity pool (The Google [organization policy](https://cloud.google.com/iam/docs/manage-workload-identity-pools-providers#restrict) needs to allow federation from the identity provider).
+- Permission to impersonate a service account needs to be granted to the external identity.
+
+Follow the detailed [instructions](https://cloud.google.com/iam/docs/access-resources-oidc) on how to configure workload identity federation from an OIDC identity provider.
+
+After configuring the OIDC provider to impersonate a service account, a credential configuration file needs to be generated.
+Unlike service account credential files, the generated credential configuration file will only contain non-sensitive metadata to instruct the library on how to retrieve external subject tokens and exchange them for service account access tokens.
+The configuration file can be generated by using the [gcloud CLI](https://cloud.google.com/sdk/).
+
+For generic OIDC providers, the Auth library can retrieve OIDC tokens either from a local file location (file-sourced credentials) or from a local server (URL-sourced credentials).
+
+**File-sourced credentials**
+For file-sourced credentials, a background process need to be continuously refreshing the file location with a new OIDC token on expiration.
```

Comment:
s/need/needs/

Also, would "prior to expiration" be more accurate?

Reply:
Done. Yes that would be more accurate. Made that change too.